presto-main/src/main/java/com/facebook/presto/operator/ExchangeClient.java

```diff
@@ -506,4 +513,27 @@ private static void closeQuietly(HttpPageBufferClient client)
             // ignored
         }
     }
+
+    private static class ExponentialMovingAverage
+    {
+        private final double alpha;
+        private double oldValue;
+
+        public ExponentialMovingAverage(double alpha, double initialValue)
+        {
+            this.alpha = alpha;
+            this.oldValue = initialValue;
+        }
+
+        public void addSample(double value)
```

Comment:
nit: I wonder if the `update` would be a better name for this method? As technically we are not adding a sample, we are updating the stored value here.

e.g.: `averageBytesPerRequest.update(value)`

Reply:
Renamed to update